parlai/core/params.py

```diff
@@ -127,22 +150,24 @@ def add_model_args(self, args=None):
             if hasattr(agent, 'add_cmdline_args'):
                 agent.add_cmdline_args(self)
             if hasattr(agent, 'dictionary_class'):
-                self.add_argument(
-                    '--dict-class', default=agent.dictionary_class(), type=str,
-                    help='the class of the dictionary agent used')
+                model_args.set_defaults(dict_class=agent.dictionary_class())
```

Comment:
I don't like it because I want to keep the options simple and save them separately, not include a class in there.
(also what happens when you deep copy the options)
Can we PLEASE change it back?